pay-api/src/pay_api/models/invoice.py

```diff
@@ -173,3 +178,43 @@ def _clean_up(self, data, many):  # pylint: disable=unused-argument
             data['status_code'] = PaymentStatus.COMPLETED.value
 
         return data
+
+
+@define
+class InvoiceSearchModel:  # pylint: disable=too-few-public-methods, too-many-instance-attributes
+    """Main schema used to serialize invoice searches, plus csv / pdf export."""
+
+    id: int
+    corp_type_code: str  # TODO probably a good idea to move this.
+    corp_type: CorpTypeSearchModel
+    created_on: datetime
+    total: Decimal
+    paid: Decimal
+    service_fees: Decimal
+    folio_number: str
+    created_name: str
+    details: List[dict]
+    payment_account: Optional[PaymentAccountSearchModel]
+    business_identifier: str
+    created_by: str
+    status_code: str
+    payment_method: str
+    line_items: Optional[List[PaymentLineItemSearchModel]]
+    product: str
+    invoice_number: str
+
+    @classmethod
+    def from_row(cls, row):
+        """From row is used so we don't tightly couple to our database class.
+
+        https://www.attrs.org/en/stable/init.html
+        """
+        return cls(id=row.id, corp_type_code=row.corp_type.code, corp_type=CorpTypeSearchModel.from_row(row.corp_type),
+                   created_on=row.created_on, total=row.total, paid=row.paid,
+                   service_fees=row.service_fees, folio_number=row.folio_number, created_name=row.created_name,
+                   details=row.details, payment_account=PaymentAccountSearchModel.from_row(row.payment_account),
+                   business_identifier=row.business_identifier, created_by=row.created_by,
+                   status_code=row.invoice_status_code, payment_method=row.payment_method_code,
+                   line_items=[PaymentLineItemSearchModel.from_row(x) for x in row.payment_line_items],
```

Comment:
Not sure if it's possible to have validators for these? What happens if payment_line_items is None for example?